api/src/main/java/jakarta/xml/bind/ContextFinder.java

```diff
@@ -309,11 +313,18 @@ static JAXBContext find(String factoryId,
             return obj.createContext(contextPath, classLoader, properties);
         }
 
-        Class<?> ctxFactory = (Class<?>) ServiceLoaderUtil.lookupUsingOSGiServiceLoader(
-                JAXBContext.JAXB_CONTEXT_FACTORY, logger);
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Iterable<Class<?>> ctxFactories = (Iterable) ServiceLoaderUtil.lookupsUsingOSGiServiceLoader(
+                JAXB_CONTEXT_FACTORY, logger);
 
-        if (ctxFactory != null) {
-            return newInstance(contextPath, contextPathClasses, ctxFactory, classLoader, properties);
+        if (ctxFactories != null) {
+            for (Class<?> ctxFactory : ctxFactories) {
```

Comment:
this is `Class<? extends JAXBContextFactory>`, right?

Reply:
Yes, but that's again staying as close as possible to the original code which was:

```
Class<?> ctxFactoryClass =
                (Class) ServiceLoaderUtil.lookupUsingOSGiServiceLoader
```

So I kept the `Class<?>` there.

Reply:
...well, either do it fully/right or keep it as it was